delfin/drivers/pure/pure/rest_handler.py

```diff
@@ -0,0 +1,150 @@
+            user = self.get_rest_login(RestHandler.REST_SESSION_URL, token,
+                                       method='POST')
+            username = user.get('username')
+            if username is None and username == '':
+                raise exception.InvalidInput('session fail to get')
+        except Exception as e:
+            LOG.error("Login error: %s", six.text_type(e))
+            raise exception.InvalidInput('login failure')
+
+    def logout(self):
+        user = self.get_rest_login(RestHandler.REST_SESSION_URL,
+                                   method='DELETE')
+        username = user.get('username')
+        if username is None and username == '':
+            raise exception.InvalidInput('delete fail to get')
+
+    def get_all_rest_volumes(self):
+        url = '%s%s:%s%s' % (
+            RestHandler.HTTPS, self.host, self.port,
+            RestHandler.REST_VOLUME_URL)
+        result_json = self.get_rest_volumes_info(url)
+        return result_json
+
+    def get_all_rest_volumes_id(self, name):
+        result_json = self.get_rest_info(
+            '{}{}'.format(RestHandler.REST_VOLUME_ID_URL, name))
+        return result_json
+
+    def get_storage(self):
+        result_json = self.get_rest_info(RestHandler.REST_STORAGE_URL)
+        return result_json
+
+    def get_storage_ID(self):
+        result_json = self.get_rest_info(RestHandler.REST_STORAGE_ID_URL)
+        return result_json
+
+    def get_all_pools(self):
+        result_json = self.get_rest_info(RestHandler.REST_POOLS_URL)
+        return result_json
+
+    def get_capacity_pools(self):
+        result_json = self.get_rest_info(RestHandler.REST_POOLS_CAPACITY_URL)
+        return result_json
+
+    def get_all_port(self):
+        result_json = self.get_rest_info(RestHandler.REST_PORT_URL)
+        return result_json
+
+    def get_all_network(self):
+        result_json = self.get_rest_info(RestHandler.REST_NETWORK_URL)
+        return result_json
+
+    def get_all_disk(self):
+        result_json = self.get_rest_info(RestHandler.REST_DISK_URL)
+        return result_json
+
+    def get_all_hardware(self):
+        result_json = self.get_rest_info(RestHandler.REST_HARDWARE_URL)
+        return result_json
+
+    def get_all_controllers(self):
+        result_json = self.get_rest_info(RestHandler.REST_CONTROLLERS_URL)
+        return result_json
+
+    def get_all_alerts(self):
+        result_json = self.get_rest_info(RestHandler.REST_ALERTS_URL)
+        return result_json
+
+    def get_rest_info(self, url, data=None, method='GET'):
+        result_json = None
+        res = self.do_call(url, data, method)
+        if res.status_code == 200:
+            result_json = res.json()
+        elif res.status_code == 401:
+            self.login()
+            self.get_rest_info(url, data, method)
+        return result_json
+
+    def get_rest_login(self, url, data=None, method='GET'):
+        self.init_http_head()
```

Comment:
`get_rest_login` is used three times, `self.init_http_head()` should just be executed only once in normal cases.